react-hot-button-component/src/index.jsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import ReactDOM from 'react-dom/client';
+
+class Button extends React.Component {
+  constructor(props) {
+    super(props);
+    this.counterValue = 0;
+    this.state = { counter: this.counterValue };
+    this.handleClick = this.handleClick.bind(this);
```

Comment:
Try reworking your solution to not add extra stuff to `this`. You should be able to rely entirely on `this.state.counter`